sp/src/game/server/ez2/npc_assassin.cpp

```diff
@@ -0,0 +1,2358 @@
+					TaskComplete();
+				}
+				break;
+			}
+
+		default:
+			BaseClass::RunTask( pTask );
+			break;
+	}
+}
+
+//---------------------------------------------------------
+//---------------------------------------------------------
+CAI_Hint *CNPC_Assassin::FindVantagePoint()
+{
+	Assert( GetEnemy() != NULL );
+
+	// Assassins following something (e.g. a commanding player) shouldn't look for a vantage point
+	if (GetFollowBehavior().CanSelectSchedule())
+		return NULL;
+
+	CHintCriteria	hint;
+
+	// Find a disadvantage node near the player, but away from ourselves
+	hint.SetHintType( npc_assassin_use_new_vantage_type.GetBool() ? HINT_TACTICAL_VANTAGE_POINT : HINT_TACTICAL_ENEMY_DISADVANTAGED );
+	hint.AddIncludePosition( GetAbsOrigin(), sk_assassin_vantage_max_dist.GetFloat() );
+	hint.AddExcludePosition( GetAbsOrigin(), sk_assassin_vantage_min_dist_from_self.GetFloat() );
+
+	Vector vecSearchPos;
+	if (GetEnemy())
+	{
+		vecSearchPos = GetEnemy()->GetAbsOrigin();
+		hint.AddExcludePosition( GetEnemy()->GetAbsOrigin(), sk_assassin_vantage_min_dist_from_player.GetFloat() );
+	}
+	else
+	{
+		// If there's no enemy, just search for any vantage point nearby
+		vecSearchPos = GetAbsOrigin();
+	}
+
+	if ( ( m_pSquad != NULL ) && ( m_pSquad->NumMembers() > 1 ) )
+	{
+		AISquadIter_t iter;
+		for ( CAI_BaseNPC *pSquadMember = m_pSquad->GetFirstMember( &iter ); pSquadMember; pSquadMember = m_pSquad->GetNextMember( &iter ) )
+		{
+			if ( pSquadMember == NULL )
+				continue;
+
+			hint.AddExcludePosition( pSquadMember->GetAbsOrigin(), sk_assassin_vantage_min_dist_from_squad.GetFloat() );
+		}
+	}
+	
+	hint.SetFlag( bits_HINT_NODE_NEAREST );
+	hint.SetFlag( bits_HINT_NODE_CLEAR );
+	hint.SetFlag( bits_HINT_NODE_USE_GROUP );
+
+	CAI_Hint *pHint = CAI_HintManager::FindHint( this, vecSearchPos, hint );
+
+	return pHint;
+}
+
+//---------------------------------------------------------
+//---------------------------------------------------------
+bool CNPC_Assassin::FValidateHintType ( CAI_Hint *pHint )
+{
+	switch( pHint->HintType() )
+	{
+		case HINT_TACTICAL_ENEMY_DISADVANTAGED:
+			if (npc_assassin_use_new_vantage_type.GetBool())
+				return BaseClass::FValidateHintType( pHint );
+		case HINT_TACTICAL_VANTAGE_POINT:
+			{
+				if (GetEnemy())
+				{
+					CBaseEntity *pEnemy = GetEnemy();
+
+					Vector	hintPos;
+					pHint->GetPosition( this, &hintPos );
+
+					// Verify that we can see the target from that position
+					hintPos += GetViewOffset();
+
+					Vector enemyPos = GetEnemies()->LastKnownPosition( pEnemy );
+					enemyPos += pEnemy->GetViewOffset();
+
+					trace_t	tr;
+					UTIL_TraceLine( hintPos, enemyPos, MASK_SHOT, this, COLLISION_GROUP_NONE, &tr );
+
+					// Check for seeing our target at the new location
+					// New vantage points require the target to be seen, old ones require the opposite
+					if ( ( tr.fraction == 1.0f ) || ( tr.m_pEnt == pEnemy) )
+						return npc_assassin_use_new_vantage_type.GetBool() ? true : false;
```

Comment:
Is this conditional problematic? Technically, if "use new vantage type" is false, but the new vantage type is the hint type, wouldn't this be the opposite condition of what should be the case for the new hint type?

Reply:
Could it be like this?
```
					if ( ( tr.fraction == 1.0f ) || ( tr.m_pEnt == pEnemy) )
						return npc_assassin_use_new_vantage_type.GetBool() && Hint->HintType() == HINT_TACTICAL_ENEMY_DISADVANTAGED;
```